src/libraries/System.Text.RegularExpressions/tests/Regex.Split.Tests.cs

```diff
@@ -57,7 +57,59 @@ public static IEnumerable<object[]> Split_TestData()
             yield return new object[] { @"(?<=\G..)(?=..)", "aabbccdd", RegexOptions.None, 8, 0, new string[] { "aa", "bb", "cc", "dd" } };
         }
 
+        // Test data addjusted from Split_TestData for RegexOptions.NonBacktracking
+        public static IEnumerable<object[]> Split_TestData_NonBacktracking()
+        {
+            yield return new object[] { "", "", RegexHelpers.RegexOptionNonBacktracking, 0, 0, new string[] { "", "" } };
+            yield return new object[] { "123", "abc", RegexHelpers.RegexOptionNonBacktracking, 3, 0, new string[] { "abc" } };
+
+            yield return new object[] { "    ", "word0    word1    word2    word3", RegexHelpers.RegexOptionNonBacktracking, 32, 0, new string[] { "word0", "word1", "word2", "word3" } };
+
+            yield return new object[] { ":", "kkk:lll:mmm:nnn:ooo", RegexHelpers.RegexOptionNonBacktracking, 19, 0, new string[] { "kkk", "lll", "mmm", "nnn", "ooo" } };
+            yield return new object[] { ":", "kkk:lll:mmm:nnn:ooo", RegexHelpers.RegexOptionNonBacktracking, 0, 0, new string[] { "kkk", "lll", "mmm", "nnn", "ooo" } };
+
+            // RegexOptions.NonBacktracking is similar to RegexOptions.ExplicitCapture when there are no explicit capture names because NonBacktracking does not support captures
+            // This is the reason for including the RegexOptions.ExplicitCapture options after the tests to check that the results are the same in this case
```

Comment:
It'd be better here to avoid duplicating all the test entries, and instead do something like:
```C#
foreach (RegexOptions options in new[] { RegexHelpers.RegexOptionNonBacktracking, RegexOptions.ExplicitCapture })
{
    ... // each yield return using options
}
```

Reply:
sounds good, will update.